include/unifex/let_error.hpp

```diff
@@ -201,167 +213,198 @@ class _frcvr<Source, Error, Receiver>::type {
 
-  template <typename Source2, typename Error2>
-  explicit type(Source2&& source, Error2&& error)
+public:
+  template <
+      template <typename...>
+      class Variant,
+      template <typename...>
+      class Tuple>
+  using value_types = type_list_nested_apply_t<
+      concat_type_lists_unique_t<
+          sender_value_type_list_t<Source>,
+          apply_to_type_list_t<
+              concat_type_lists_unique_t,
+              map_type_list_t<final_senders_list, sender_value_type_list_t>>>,
+      Variant,
+      Tuple>;
+
+  template <template <typename...> class Variant>
+  using error_types = typename concat_type_lists_unique_t<
+      sender_error_type_list_t<Source>,
+      apply_to_type_list_t<
+          concat_type_lists_unique_t,
+          map_type_list_t<final_senders_list, sender_error_type_list_t>>,
+      type_list<std::exception_ptr>>::template apply<Variant>;
+
+  static constexpr bool sends_done =
+    sender_traits<Source>::sends_done ||
+    sender_error_types_t<Source, sends_done_impl>::value;
+
+  template <typename Source2, typename Func2>
+  explicit type(Source2&& source, Func2&& func)
     noexcept(
       std::is_nothrow_constructible_v<Source, Source2> &&
-      std::is_nothrow_constructible_v<Error, Error2>)
+      std::is_nothrow_constructible_v<Func, Func2>)
     : source_((Source2&&)source)
-    , error_((Error2&&)error)
+    , func_((Func2&&)func)
   {}
 
   template(
     typename Sender,
     typename Receiver,
     typename...,
-    typename SourceReceiver = receiver_type<member_t<Sender, Source>, Error, Receiver>,
-    typename FinalReceiver = final_receiver_type<member_t<Sender, Source>, Error, Receiver>)
+    typename SourceReceiver = receiver_type<member_t<Sender, Source>, Func, Receiver>)
       (requires same_as<remove_cvref_t<Sender>, type> AND
-          constructible_from<Error, member_t<Sender, Error>> AND
+          constructible_from<Func, member_t<Sender, Func>> AND
           constructible_from<remove_cvref_t<Receiver>, Receiver> AND
-          sender_to<member_t<Sender, Source>, SourceReceiver> AND
-          sender_to<final_sender_t, FinalReceiver>)
+          sender_to<member_t<Sender, Source>, SourceReceiver>)// AND
+          // sender_to<final_sender_t, FinalReceiver>)
   friend auto tag_invoke(tag_t<unifex::connect>, Sender&& s, Receiver&& r)
        noexcept(
         is_nothrow_connectable_v<member_t<Sender, Source>, SourceReceiver> &&
-        std::is_nothrow_constructible_v<Error, member_t<Sender, Error>> &&
+        std::is_nothrow_constructible_v<Func, member_t<Sender, Func>> &&
         std::is_nothrow_constructible_v<remove_cvref_t<Receiver>, Receiver>)
-      -> operation_type<member_t<Sender, Source>, Error, Receiver> {
-    return operation_type<member_t<Sender, Source>, Error, Receiver>{
+      -> operation_type<member_t<Sender, Source>, Func, Receiver> {
+    return operation_type<member_t<Sender, Source>, Func, Receiver>{
```

Comment:
I think you should remove `member_t` here so that the operation is instantiated with a non-reference type on `Source`.
```suggestion
    return operation_type<Source, Func, Receiver>{
```